### PR TITLE
Support Geant4-VecGeom conversion of parameterized volumes

### LINKED ISSUE #1068: Add G4VG support for ALICE Geant4 geometry components
From @sawenzel:

> For the moment, I seem to be blocked by 2 error messages:
> 
> a) unsupported solid type G4DisplacedSolid
> 
>  b) unsupported type 'G4PVDivision' for physical volume 'ZNTX_PV' (corresponding LV: "ZNTX"@0xaaaabefaf6d0 (ID=106))
> 
> - G4DisplacedSolids are sometimes used inside booleans. Not sure why you don't see this with CMS but
> it could be a speciality of the GDML constructed with ROOT.  We would need to implement the solid in VecGeom first and ensure it works on CUDA.
> 
> - The other issue about G4PVDivision can probably be solved in the geometry converter by simply creating distinct VecGeom physical volumes or maybe there is an option in the ROOT geometry exporter not to use divisions in the output... I'll take a look.

From discussions with the VecGeom team it sounds like we might be able to reduce the displaced solids to regular solids by propagating the displacement into the "placement" of the physical volumes.

For the physical volumes, we should be able to replicate them manually.

### PR DESCRIPTION
**Pull request by sethrj**
This adds support for parameterized Geant4 volumes. Closes #1068 .

### CODE REVIEW

File: src/celeritas/ext/g4vg/Converter.cc [hunk 1]
```diff
@@ -159,45 +230,56 @@ auto Converter::build_with_daughters(G4LogicalVolume const* mother_g4lv)
...
-        vecgeom::ReflFactory::Instance().Place(
-            (*this->convert_transform_)(g4pv->GetTranslation(),
-                                        g4pv->GetRotation()),
-            vecgeom::Vector3D<double>{1.0, 1.0, flip_z ? -1.0 : 1.0},
-            g4pv->GetName(),
-            daughter_lv,
-            mother_lv,
-            g4pv->GetCopyNo());
+            // Loop over number of replicas
+            for (auto j : range(div->GetMultiplicity()))
```
Comment (sethrj) [inline src/celeritas/ext/g4vg/Converter.cc]: @sawenzel You used `g4pv->GetReplicationData` but while digging into the replication stuff this morning, it looked like this was the method more frequently used. Is there a subtle difference?

File: test/celeritas/ext/Vecgeom.test.cc [hunk 2]
```diff
@@ -1460,6 +1472,33 @@ TEST_F(SolidsGeantTest, reflected_vol)
...
+    // an ALICE component
+    static char const* const expected_mid_volumes[]
+        = {"World", "ZNENV", "ZNST", "ZNST",  "ZNST", "ZNST", "ZNST",
+           "ZNST",  "ZNST",  "ZNST", "ZNST",  "ZNST", "ZNST", "ZNST",
+           "ZNST",  "ZNST",  "ZNST", "ZNST",  "ZNST", "ZNST", "ZNST",
+           "ZNST",  "ZNST",  "ZNST", "ZNENV", "World"};
+    static real_type const expected_mid_distances[]
+        = {6.38, 0.1,  0.32, 0.32, 0.32, 0.32, 0.32, 0.32, 0.32,
+           0.32, 0.32, 0.32, 0.32, 0.32, 0.32, 0.32, 0.32, 0.32,
+           0.32, 0.32, 0.32, 0.32, 0.32, 0.32, 0.1,  46.38};
```
Comment (sethrj) [inline test/celeritas/ext/Vecgeom.test.cc:1487]: These results are from the geant4 navigator; the geometry is a subset of the ZDC input that has the division volumes.

File: src/celeritas/ext/g4vg/Converter.cc [hunk 3]
```diff
@@ -159,45 +230,56 @@ auto Converter::build_with_daughters(G4LogicalVolume const* mother_g4lv)
...
-            flip_z = true;
+            DaughterPlacer place_daughter(convert_daughter,
+                                          *convert_transform_,
+                                          g4pv->GetLogicalVolume(),
+                                          mother_lv);
+
+            // Place daughter, accounting for reflection
+            place_daughter(placed);
         }
+        else if (auto* div = dynamic_cast<G4PVDivision*>(g4pv))
```
Comment (sawenzel) [inline src/celeritas/ext/g4vg/Converter.cc]: looking good to me overall. Maybe one could generalize from G4PVDivision to G4PVReplica ?(although I am no real expert on this).

**Review by sawenzel — APPROVED** [review]

File: src/celeritas/ext/g4vg/Converter.cc (hunk 1 above)
Comment (sawenzel) [reply src/celeritas/ext/g4vg/Converter.cc]: No. I was simply not aware if this leaner API.

File: src/celeritas/ext/g4vg/Converter.cc (hunk 3 above)
Comment (sethrj) [reply src/celeritas/ext/g4vg/Converter.cc]: Looks like "replica" doesn't necessarily mean "parameterized", but that's a good idea so I've generalized it a bit.

**Review by amandalund — APPROVED** [review]
Looks good, thanks @sethrj!

**Pull request merged**